category form UI: add button

xi opened the pull request:
This is very similar to #4, but I added an "add" button and fixed some of the keyboard related issues I described in https://github.com/kub-berlin/kub-weiterleitungsliste/pull/4#issuecomment-953979714.

I like the idea of encouraging users to re-use existing categories and only show the text fields on explicit request.

To make sure that at least one category is selected I added an error message (see b4f3da8)

xi committed:
add category instead of submit on enter

xi committed:
define state.formCategories

xi committed:
rm compat code

xi committed:
display formCategories in form

xi committed:
allow to remove formCategory

xi committed:
error on missing category

xi committed:
allow to add categories

ancoulibaly commented:
I like it! There are just two issues that I encountered while testing it:

1. When you create a new category and add it with the add button, the two category fields are still displayed, although now empty. I would expect them to disappear again.

2. When you try to submit the form without choosing a category you get a warning message, but once you click on "ok", the add button is disabled and you cannot add any category.

And since it is possible to add not only new subcategories but also new categories, should we add a general "new" option at the end of the `select` element's dropdown list?

ancoulibaly commented on static_src/main.js:
```diff
@@ -131,15 +131,15 @@ var onNavigate = function(event, state) {
     }
 
+    // prevent double-submit
+    var submit = event.target.querySelector('button:not[form]');
```
Brackets are missing. It should read: `'button:not([form])'`

ancoulibaly commented:
Thanks for taking care of the issues! I just noticed that you can select an option in the `select` element via the keyboard, but it will not be added when pressing enter. Is this intended?

xi commented:
> Thanks for taking care of the issues! I just noticed that you can select an option in the select element via the keyboard, but it will not be added when pressing enter. Is this intended?

Yes, that is part of how HTML works. We should not mess with that.

xi committed:
split categoryOptions()

xi committed:
sort categories before upload

xi committed:
fix select height

xi committed:
allow to create new parent category

To simplify the code I used petit-dom's feature to automatically flatten
lists of elements.
